dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/TrackerPreheat.java

```diff
@@ -133,6 +141,122 @@
     @Getter
     private Map<String, PeriodType> periodTypeMap = new HashMap<>();
 
+    /**
+     * Internal map of category combo + category options (key) to category
+     * option combo (value).
+     *
+     * Category option combo value will be in the idScheme defined by the user
+     * on import.
+     */
+    private final Map<Pair<String, Set<String>>, String> cosToCOC = new HashMap<>();
+
+    /**
+     * Store mapping of category combo + category options identifiers(key) to
+     * category option combo identifiers (value).
+     *
+     * Category options, category option combo identifiers will be in the
+     * idScheme defined by the user on import. Note: different idSchemes for
+     * category combos are not supported.
+     */
+    public void putCategoryOptionCombo( CategoryCombo categoryCombo, Set<CategoryOption> categoryOptions,
```

Comment:
@enricocolasante I struggled a bit with naming, maybe you have some thoughts.

So we had this map `cosToCOC` before in the validation hook. Now its in the preheat, a bit more robust for when it comes to payload inputs (like ` id1;id1 `, `id2;id1`, `id1;id2`) since I use a set of categoryOption identifiers. So duplicates and order does not matter, also trimming whitespace.

What I wonder is if its confusing that there can be a case in which

`preheat.getCategoryOptionCombo(id)` returns a CategoryOptionCombo
`preheat.getCategoryOptionCombo(categoryCombo, categoryOptions)` returns null

The reason being is that the ClassBasedSupplier will add all CategoryOptionCombo for which there are ids in the payload into the preheat `map`.

The newly added `EventCategoryOptionComboSupplier` will add only CategoryOptionCombo for event payloads that have no CategoryOptionCombo id but CategoryOptions to the preheat `map` and `cosToCOC`. This supplier calls `putCategoryOptionCombo`.

So if

`preheat.getCategoryOptionCombo(categoryCombo, categoryOptions)` returns a CategoryOptionCombo
`preheat.getCategoryOptionCombo(id)` will return the same CategoryOptionCombo

Bottom line, do you think the names are confusing since they are the same, just the signature is different? I added javadocs that explain all of the above. I could not think of a different name. Not sure if that would be necessary.

Reply:
I think it makes sense to have the same name for overloaded methods.
In the end you are retrieving the same thing just using different parameters.